### PR TITLE
Added Additional JSDoc Typechecking

### PR DESCRIPTION
**Pull request by kbader-godaddy**
## Summary

Added typechecking to the following plugins:
- `@gasket/plugin-docusaurus`
- `@gasket/plugin-elastic-apm`
- `@gasket/plugin-fastify`
- `@gasket/gasket-utils`
- `@gasket/plugin-lint`

## Test Plan

All existing tests pass with the included `typecheck` scripts.

### CODE REVIEW

File: packages/gasket-utils/lib/apply-config-overrides.js [hunk 1]
```diff
@@ -1,49 +1,71 @@
...
   );
 }
 
+/**
+ * Generator function to yield potential configurations
+ * @type {import('./index').getPotentialConfigs}
+ */
 function *getPotentialConfigs({ config, env, commandId, root, localFile }) {
   // Separate environment-specific config from other config
+  // @ts-ignore
```
Comment (agerard-godaddy) [inline packages/gasket-utils/lib/apply-config-overrides.js:30]: Why all the `@ts-ignores` in this function?

File: packages/gasket-utils/lib/package-manager.js [hunk 2]
```diff
@@ -78,19 +70,19 @@ class PackageManager {
...
       const argv = [
-        '--loglevel', 'info',
-        '--spin', 'false',
-        '--progress', 'false'
+        '--loglevel',
+        'info',
+        '--spin',
+        'false',
+        '--progress',
+        'false'
```
Comment (agerard-godaddy) [inline packages/gasket-utils/lib/package-manager.js:83]: This was easier to understand with flag/value in same line

**Review by mmason2-godaddy — APPROVED** [review]

File: packages/gasket-utils/lib/package-manager.js (hunk 2 above)
Comment (kbader-godaddy) [reply packages/gasket-utils/lib/package-manager.js:83]: Great feedback! Will update!

File: packages/gasket-utils/lib/apply-config-overrides.js (hunk 1 above)
Comment (kbader-godaddy) [reply packages/gasket-utils/lib/apply-config-overrides.js]: This was included by mistake. Just pushed the update.

**Pull request merged**